### PR TITLE
Refactor: Chat using graphql and pagination

### PR DESCRIPTION
**Pull request by Tainan404**
### What does this PR do?

This PR introduces the graphql to the chat messages in BBB, with the new pagination we stabilzed pages and load it incrementaly as the user request to look into the message history, not storing more the whole chat history in client memory.

### Motivation

Make the chat support graphql and make use of pagination to improve client performance

### CODE REVIEW

**Comment by gustavotrott** [comment]
image (2)

It seems that it is missing one message between two pages.
Maybe you need to tweak the offset.

**Comment by gustavotrott** [comment]
https://github.com/bigbluebutton/bigbluebutton/assets/5660191/a0848fff-8a57-40f4-a8fb-b533331e1703

When I scroll down a little, the scroll bar jumps directly to bottom.

**Comment by gustavotrott** [comment]
https://github.com/bigbluebutton/bigbluebutton/assets/5660191/4e99afd3-9a2e-41b2-bea3-4f082c0ec8ee

I'm stopped at the top, and other users are sending msg.
When a new page is created, my current page changes (while I'm reading it..).

File: bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/component.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        </div>
+      </MessageList>
+    </MessageListWrapper>
+  );
+}
+
+const ChatListContainer: React.FC = ({}) => {
+  const idChatOpen = layoutSelect((i) => i.idChatOpen);
+  const isPublicChat = idChatOpen === PUBLIC_CHAT_KEY;
+  const chatId = !isPublicChat ? idChatOpen : PUBLIC_GROUP_CHAT_KEY;
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/component.tsx:180]: For a reduced cognitive effort, please change it to:
```
  const chatId = isPublicChat ? PUBLIC_GROUP_CHAT_KEY : idChatOpen;
```

File: bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/component.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  const {
+    data: chatData,
+    loading: chatLoading,
+    error: chatError,
+  } = useSubscription<ChatSubscriptionResponse>(CHAT_SUBSCRIPTION);
+
+  if (chatError) return <p>chatError: {chatError}</p>;
+  const currentChat = chatData?.chat?.filter((chat) => chat?.chatId === chatId)?.[0];
```
Comment (TiagoJacobs) [inline bigbluebutton-html5/imports/ui/components/chat/chat-graphql/chat-message-list/component.tsx:188]: Use `find` instead `filter()[0]`.

Example:

```js
[1,2,3,4,5].find(function (el) {
    return el > 3;
});
```

**Review by TiagoJacobs — CHANGES_REQUESTED** [review]
Hello! 

I propose that we remove the old meteor code in this PR (instead of just adding the graphql code).

The old code will always be available in git history / 2.6.x branches.

**Comment by gustavotrott** [comment]
When I just open the chat, it always loads the page 1.
And then immediately remove it and load the last and last-1 pages.

https://github.com/bigbluebutton/bigbluebutton/assets/5660191/f88c1da4-ca89-4961-8e88-19b690fb80bb

I think it shouldn't waste bandwidth downloading the first page if it will not be shown.
And it also shows a weird switch of content.

### - Second 00:46 shows message 50
image

### - Second 00:47 hides that page
image

### - Second 00:48 shows message 1012
image

**Review by gustavotrott — APPROVED** [review]
lgtm! great job, the chat is really seamless in this new version!

**Pull request merged**